robotframework-ls/src/robotframework_ls/impl/semantic_tokens.py

```diff
@@ -136,33 +129,19 @@ def _tokenize_token(node, initial_token):
             initial_token_type = KEYWORD
 
     if initial_token_type == KEYWORD:
-        dot_pos = initial_token.value.rfind(".")
-        if dot_pos > 0:
-            tok = _DummyToken()
-            tok.type = "name"
-            tok.value = initial_token.value[:dot_pos]
-            tok.lineno = initial_token.lineno
-            tok.col_offset = initial_token.col_offset
-            prev_col_offset_end = tok.end_col_offset = initial_token.col_offset + len(
-                tok.value
-            )
-            yield tok, TOKEN_TYPE_TO_INDEX["name"]
-
-            # tok = _DummyToken()
-            # tok.type = "control"
-            # tok.value = "."
-            # tok.lineno = initial_token.lineno
-            # tok.col_offset = prev_col_offset_end
-            prev_col_offset_end = tok.end_col_offset = prev_col_offset_end + 1
-            # yield tok, TOKEN_TYPE_TO_INDEX["control"]
-
-            tok = _DummyToken()
-            tok.type = initial_token_type
-            tok.value = initial_token.value[dot_pos + 1 :]
-            tok.lineno = initial_token.lineno
-            tok.col_offset = prev_col_offset_end
-            tok.end_col_offset = prev_col_offset_end + len(tok.value)
-            yield tok, RF_TOKEN_TYPE_TO_TOKEN_TYPE_INDEX[initial_token_type]
+        token_to_process = _DummyToken(initial_token)
```

Comment:
Methods from dummy class are (now) used to extract gherkin prefix and module prefix. So this allocation is needed

Reply:
I was actually thinking that instead of creating the instance to do the match, it could be nicer to not add the methods related to parsing to the `_DummyToken` and just make those top-level (for me having those methods as a part of the `_DummyToken` seems a bit weird... the `_DummyToken` should keep on being a token and maybe having manipulation methods, so, something as `extract_token` makes sense to be there, but having the knowledge about gherkin/library seems to be a bit misplaced for me).

-- one of the guidelines https://github.com/robocorp/robotframework-lsp/blob/master/robotframework-ls/docs/contributing.md#codebase-styleguidelines is indeed that `"... most of the code should reside in the same module and functions are usually preferred over classes -- but not always ;)."`

IMHO, this leads to less structuring/architecting and usually makes the code simpler to follow.

(pseudo-code below)

```
# Note: top-level method instead of _DummyToken instance method
def extract_gherkin_token_from_keyword(token):
    match = gherking_regexp.match(token.value)
    if match:
        if token.__class__ != _DummyToken:
            token = _DummyToken(initial_token)
        gherkin_token = token.extract_token("control", match.end())
        return token, gherkin_token
    return token, None

def extract_library_token_from_keyword(token):
    dot_pos = token.value.rfind(".")
    if dot_pos > 0:
        if token.__class__ != _DummyToken:
            token = _DummyToken(initial_token)
        library_token = token.extract_token("name", match.end())
        return token, library_token
    return token, None

...

token, token_gherkin_prefix = extract_gherkin_token_from_keyword(initial_token)
if token_gherkin is not None:
    yield token_gherkin_prefix, TOKEN_TYPE_TO_INDEX[token_gherkin_prefix.type]
token, library_token = extract_library_token_from_keyword()
if library_token is not None:
    yield library_token, TOKEN_TYPE_TO_INDEX[library_token.type]
```

Reply:
I can live with that